### PR TITLE
handle ARG instruction in Dockerfile

### PR DESCRIPTION
**Pull request by reidpr**
This extends PR #490 by @cmbiwer and also addresses issue #501.

The extensions are:

* generalize `ARG` and `--build-arg` syntax per Docker and Buildah documentation
* add tests
* document `--build-arg`

The one reduction in functionality is to accept exactly one `KEY[=VALUE]` per `--build-arg`. My reading of both the Docker and Buildah docs is that they have this behavior. Also, if we consume any number of variables, constructions like `ch-grow [...] --build-arg foo=bar baz` will consume `baz` as a build argument when it's intended as the context directory.

@cmbiwer, does this extended PR meet your needs? I apologize for creating a duplicate PR; I can't push to your branch.

### CODE REVIEW

File: bin/ch-grow [hunk 1]
```diff
@@ -505,12 +544,16 @@ def INFO(*args, **kwargs):
...
    DEBUG("executing: %s" % args)
-   INFO("\033[33m", end="")
+   color("33m", sys.stdout)
    cp = subprocess.run(args, env=env, stdin=subprocess.DEVNULL)
-   INFO("\033[0m", end="")
+   color("0m", sys.stdout)
    if (cp.returncode):
       FATAL("%s failed with return code %d" % (args[0], cp.returncode))
 
+def color(color, fp):
```
Comment (rstyd) [inline bin/ch-grow]: Should the color stuff technically be in a different commit? I'm fine with keeping it here.

Comment (reidpr) [reply bin/ch-grow]: Remember we squash-merge, so this will all become one commit.

You're right that it's a second concern and would be tidier in another PR. However, `ch-grow` output can't be tested without fixing #501, which is why it's here.

**Comment by cmbiwer** [comment]
@reidpr Yes, looks like this PR still meets my use case. Thanks!

File: bin/ch-grow (hunk 1 above)
Comment (rstyd) [reply bin/ch-grow]: Sounds good to me!

**Comment by rstyd** [comment]
Approved.
